app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/SubsidyController.scala

```diff
@@ -198,11 +207,14 @@ class SubsidyController @Inject()(
       claimEoriForm.bindFromRequest().fold(
         formWithErrors => Future.successful(BadRequest(addClaimEoriPage(formWithErrors, previous))),
         form => {
-          store.update[SubsidyJourney]({ x =>
-            x.map { y =>
-              y.copy(addClaimEori = y.addClaimEori.copy(value = Some(form.value.map(EORI(_)))))
-            }
-          }).flatMap(_.next)
+          for {
+            journey <- store.update[SubsidyJourney]({ x =>
```

Comment:
If you are going to use explicit variable names then it might be better to give them meaningful names, since on the face of it, its not obvious what `x` and `y` are here. Also looking at this I can't help but think there has to be a better way to do this, but happy to leave that for later.